OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -138,8 +158,22 @@ void DisguiseAs(Actor self, Actor target)
 
 			foreach (var t in self.TraitsImplementing<INotifyEffectiveOwnerChanged>())
 				t.OnEffectiveOwnerChanged(self, oldEffectiveOwner, AsPlayer);
+
+			if (Disguised != oldDisguiseSetting && um.Value != null)
+			{
+				foreach (var u in info.Upgrades)
+				{
+					if (!um.Value.AcceptsUpgrade(self, u))
+						continue;
+
+					if (Disguised)
```

Comment:
Turned out that's not the issue. Due to UpgradeMaxAcceptsLevel defaults to 1, the AcceptsUpgrade check notices the Spy cannot taker another Disguise upgrade and returns false even before this line.